### DISCUSSION TITLE
Getting started

### DISCUSSION THREAD

**Discussion by mscottford** [ImoutoChan/openmoq#1] [General]
I'd love to dig in and help with this effort. I _almost_ started my own fork before I found this one. I do have some questions, though. Hopefully, they are seen as welcome and not overwhelming. 😅 

1. Are you open to collaboration on this effort? Like I said. I'm willing to help.
    * What form should that collaboration take? I could get started by creating some issues and pull requests. 
2. Do you plan on tracking changes from the upstream project? My preference for creating my own fork was to completely break from the original, including resetting the version number down to 1.0.0. I do see that you've already published version 4.20.1, so perhaps not on the version number. 😄 
    * If yes on breaking from the upstream project, then I think it would also make sense to reach out to GitHub's support to disconnect the GitHub repository from the Moq project. What are your thoughts on that?
4. Do you plan on updating the codebase to remove the name `Moq` anywhere that it appears, outside of a short mention in the README and the LICENCE? That was something that I was going to do with the fork that I had in mind.
    * If this is the direction that you want to go in, then the logo should also be removed/changed. 
5. Are you open to moving the project to a separate GitHub and NuGet organizations, so that the project does not live under your namespace?

**Comment by mscottford** [comment]
@ImoutoChan have you had a chance to think about my questions?

**Reply by ImoutoChan** [reply]
Hi!

Unfortunately, I didn't receive any notifications about this thread before @ mention, so I'm going to answer it only now. I'm glad to see new faces on this project!

1. I will welcome any help! You can start with issues and we see what we can do about them. If you see any mistakes or want to clean something up you are welcome to create pr right away.
2. My original intention was to keep it up to date with the upstream with the exclusion of certain changes. What benefits can we get after breaking from the original repo? About the version, I want people to see migrating to this project as some kind of update, where you can bulk replace "moq" reference with the "openmoq" one.
3. My concern here is the license. I guess we at least have to save the author's name, and I'm not sure about the original name of the project. If the license allows us to rename everything, I'm ok with accepting pr where every moq is renamed to openmoq.
4. I guess I'm ok with that. Originally I wasn't so sure someone else was going to reach out and help. Should I create an organization named openmoq and move this repo there?

**Reply by ImoutoChan** [reply]
I also thought that the author of the library would change his license in the near future. After that, we will be forced to break from the upstream. But I don't see any harm in leaching his changes, bugfixes, and features before that time come.
